bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/RuntimeUtils.java

```diff
@@ -196,4 +198,66 @@ private static String initBRELogHandler() {
             return Paths.get(TEMP_DIR, fileName).toString();
         }
     }
+
+    /**
+     * Check for the compatibility of a given java version against the current java runtime version.
+     * This assumes the versions are in the following formats:
+     * <ul>
+     * <li>Java 8 or below: <i>1.8.x</i>, <i>1.7.x</i>, <i>1.6.x</i>, etc.</li>
+     * <li>Java 9 or above: <i>9.x.y.z</i>, <i>10.x.y.z</i>, <i>11.x.y.z</i>, etc.</li>
+     * </ul>
+     * 
+     * @param compiledVersion java version to check the compatibility
+     */
+    public static void verifyJavaCompatibility(String compiledVersion) {
+        String runtimeVersion = System.getProperty(JAVA_VERSION);
+        if (compiledVersion == null || runtimeVersion == null || compiledVersion.equals(runtimeVersion)) {
+            return;
```

Comment:
Can compiledVersion or runtimeVersion be null ?  then don't we need to log an error?

Reply:
Ideally it cannot be null. But added the check for the null-safety.